src/Orleans/Core/GrainClient.cs

```diff
@@ -21,6 +21,13 @@ namespace Orleans
     public delegate void ConnectionToClusterLostHandler(object sender, EventArgs e);
 
     /// <summary>
+    /// The delegate called before every request to a grain.
+    /// </summary>
+    /// <param name="request">The request.</param>
+    /// <param name="grain">The grain.</param>
+    public delegate void ClientInvokeCallback(InvokeMethodRequest request, IGrain grain);
```

Comment:
Yeah, I was looking at a commit where this was defined twice simultaneously, not just a move. But a later commit removed one of them, so it's OK